### ISSUE TITLE
Saving OBJ files not working

### ISSUE THREAD

**Issue by konsumer** [notnullgames/Herosaver#1]
Currently STL & JSON bookmarklets work, but OBJ stopped working.

In dev-console, I get a bunch of errors, like this:

```
THREE.OBJExporter.parseMesh(): geometry type unsupported 
```

**Comment by konsumer** [comment]
The exported STL appears valid, but blender crashes when I try to import it, so it might also be damaged.

**Comment by chiefofkaiser** [comment]
Opening the Exported STL in 3D Builder, Cura, or Blender displays a large spikey mass.  OBJ file does not open.  Let me know if there is anything else I can add

**Comment by konsumer** [comment]
I think I need to look at the JSON, when I have more time, and see if it's good. if so, then they might be doing soemthing to break the output.

**Comment by CodilX** [comment]
They did make changes, and substantial ones at that. When loading meshes and after (from what I can tell) initial transformations, they obfuscate the meshes by changing weights, morphs, etc to pseudo random absurd values and those meshes get loaded into memory. The process is reversed when preparing the model for rendering. That _modified_ mesh is what we can access and is why we're getting a garbled mess (take all of this with a grain of salt, it's non trivial to read and follow minified code even when beautified).

To bypass this requires modifying their functions to terminate before obfuscation or by undoing the obfuscation process when exporting. 

The issue at hand is this would essentially mean breaking/cracking their "protection"/drm/whatever you want to call it. Up till now all iterations of "herosaver" were a purely academic exercise hovering in a gray area, but "cracking" their obfuscation would surely cross the line into legal trouble territory (speculation).

**Comment by lillypath869** [comment]
Personally, I like what they are doing with the upper body. Basically making it a solid cube that engulfs the entire bounding box of the model. 

From what I had seen, they are only messing with weights and not morphs. 

Lastly, they put defensive measures in for other implementations of the script that modify the DOM to add menus. For those, they are sorting the geometry position attributes.

**Comment by Antharon** [comment]
so there are two ways to fix this - either bypass that obfuscation by setting all those weights to neutral number or collecting them and applying them in reverse?

When I first ripp-off herosaver I put saving script into chrome extension, so script will not inject any elements into DOM, this might help with preventing detection of herosaver. You can put whole herosaver UI into extension popup.

**Comment by NilsBaumgartner1994** [comment]
leaving dot for new informations

**Comment by konsumer** [comment]
@Antharon I'm not sure how to do either of your suggestions, but I'll accept a PR. RE: extension: I'd prefer to not need an extension that needs to be rewritten for each browser. Greasemonkey can do the same stuff, if we really need something that runs outside the context of the page, but I have a feeling we don't need that.

**Comment by jesmartinez** [comment]
Any news on the project? I tried to fix it without success.

**Comment by CodilX** [comment]
In the latest update - a huge cube envelopes the exported model that is in a T pose with some meshes wrongly positioned. They also redid the weights and indexes, but maybe someone can figure it out ;)

**Comment by konsumer** [comment]
I'm pretty busy right now, so I don't have time to dig in, but I'm happy to accept PRs if anyone else works it out.

**Comment by srgank666** [comment]
HI there. There is no news on this one? I'm trying to take a look but I'm javascript newbie....

**Comment by konsumer** [comment]
Nope. I haven't had any time, and I'm not sure it's fixable, to begin with. I'll happily accept a PR, if someone figures it out, though.

**Comment by LonelyDeveloper0** [comment]
Any news? Did you manage to fix it? I really need this .obj option, export the model with color

**Comment by adunis** [comment]
Hello!
No, I didn't manage to fix it.

**Comment by LonelyDeveloper0** [comment]
really sad, do you know another way to do this?
